### PR TITLE
Fixed proxy status check on sn-platform

### PR DESCRIPTION
**Pull request by tuteng**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

```
auth:
  authentication:
    statusCheck: true
```

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: charts/sn-platform-slim/values.yaml [hunk 1]
```diff
@@ -305,6 +305,7 @@ auth:
       autoInitSecret: false
     tls:
       enabled: false
+    statusCheck: false
```
Comment (yaalsn) [inline charts/sn-platform-slim/values.yaml:308]: Why do we set its default value `false` ?

**Review by yaalsn — APPROVED** [review]

File: charts/sn-platform-slim/values.yaml (hunk 1 above)
Comment (ericsyh) [reply charts/sn-platform-slim/values.yaml:308]: @tuteng pls update the default value to `true`

**Review by yaalsn — APPROVED** [review]
LGTM, thanks!

**Comment by ericsyh** [comment]
If Pulsar will fix this issue in later version, maybe we can just ask users to config in the https://github.com/streamnative/charts/blob/master/charts/sn-platform-slim/templates/proxy/proxy-cluster.yaml#L209 rather integrate this in the template. @tuteng @yaalsn

**Comment by tuteng** [comment]
Close this pr, add the following configuration to the value file proxy.customization section to enable authentication configuration for status.html

```
proxy:
  customization:
  - manifest: |
      spec:
        template:
          spec:
            $setElementOrder/containers:
            - name: pulsar-proxy
            containers:
            - name: pulsar-proxy
              livenessProbe:
                httpGet: null
                exec:
                  command:
                  - /bin/bash
                  - -c
                  - 'res=$(curl -H "Authorization: Bearer $brokerClientAuthenticationParameters" http://localhost:8080/status.html);if [[ $res == "OK" ]]; then exit 0; else exit 1; fi'
              readinessProbe:
                httpGet: null
                exec:
                  command:
                  - /bin/bash
                  - -c
                  - 'res=$(curl -H "Authorization: Bearer $brokerClientAuthenticationParameters" http://localhost:8080/status.html);if [[ $res == "OK" ]]; then exit 0; else exit 1; fi'
              startupProbe:
                httpGet: null
                exec:
                  command:
                  - /bin/bash
                  - -c
                  - 'res=$(curl -H "Authorization: Bearer $brokerClientAuthenticationParameters" http://localhost:8080/status.html);if [[ $res == "OK" ]]; then exit 0; else exit 1; fi'
    match:
      groupVersionKinds:
      - kind: StatefulSet
      name: .*-proxy
```

**Pull request closed without merging**